Shipping Labels: order summary and woo discount section

pmusolino opened the pull request:
Part of #4088 

## Description
As part of Shipping Labels M3, in this PR I implemented the order summary that displays the Subtotal, the WooCommerce discount (if available), the Order Total, a switch for completing the order, and the button for purchasing the shipping label.
I've implemented a new cell `ShippingLabelSummaryTableViewCell` that displays all the information needed when the user fulfilled all the fields in the Shipping Labels form.

## Testing
1. Make sure you have configured the WooCommerce WCShip plugin, and you have one or more payment methods set up in WP Admin under WooCommerce > Settings > Shipping > WooCommerce Shipping.
2. Launch the app in debug mode (because of the feature flag).
3. Open an order detail that is eligible for the shipping label.
4. Tap the button "Create Shipping Label".
5. Complete all the fields, and try to select the more expensive option under "Shipping Carriers and Rates" (in theory, this makes your shipping label eligible for our WooCommerce discount).
6. You should be able to see the "Shipping Label Order Summary" section at the end of the Shipping Label Form, and the "WooCommerce Discount" row should be visible.
7. Try to change the "Shipping Carriers and Rates" option, selecting the cheapest one. -> You shouldn't be able to see anymore the discount row.

PS: There is no guarantee that the cheapest carrier will not guarantee a discount, and the reverse is also true, an expensive carrier does not guarantee a discount. So do several tests if you are not able to see the discount row.


## Screenshots
| Order Summary with discount             |  Order Summary without discount |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 12 Pro - 2021-06-15 at 16 53 45](https://user-images.githubusercontent.com/495617/122075753-99c0ce80-cdfa-11eb-8569-ed358e464871.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-06-15 at 16 55 48](https://user-images.githubusercontent.com/495617/122075774-9d545580-cdfa-11eb-9439-203bd05b3f6a.png)




Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

pmusolino committed:
feat: implemented a first version of ShippingLabelSummaryTableViewCell

pmusolino committed:
feat: header config

pmusolino committed:
feat: handle Shipping Label Summary content

pmusolino committed:
fix: cell with dark mode enabled

pmusolino committed:
Merge branch 'develop' into issue/4088-view-order-summary-and-woo-discount

rachelmcr commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift:
```diff
@@ -224,6 +224,73 @@ final class ShippingLabelFormViewModel {
+            return nil
+        }
+
+        var rate: Double = selectedRate.retailRate - selectedRate.rate
```
I noticed that the discount appears as a positive number if I select an expensive shipping rate without adding any signature option. It looks like `retailRate` and `rate` should be switched here:

```suggestion
        var rate: Double = selectedRate.rate - selectedRate.retailRate
```

pmusolino replied:
Good catch!

rachelmcr reviewed (commented):
For the most part, this looks and works great! I confirmed the discount line is shown/hidden as expected, and the summary section only appears when all of the form sections are complete.

There's a small issue with the WooCommerce discount showing as a positive value when a signature option isn't selected for the shipping rate (see suggested code change below).

I also noticed a couple issues with accessibility and localization, but we could probably address those separately. (We should probably add a task to the Shipping Labels M3 project to test things like accessibility and localization through the whole flow; what do you think?) Here's what I noticed:

* Accessibility: With large font sizes, "WooCommerce Discount" gets cut off. (I also noticed that with _really_ large font sizes, the discounted amount gets pushed off the screen.) Should the titles wrap when they don't fit on one line?

<img src="https://user-images.githubusercontent.com/8658164/122101825-a8b07c80-ce0c-11eb-98da-900750792de7.png" width="300px">

* Localization: With right-to-left languages (such as Arabic), the titles aren't aligned correctly in the summary cell rows. Looking just at the fully translated rows, you can see that the titles are left-aligned instead of right-aligned:

<img src="https://user-images.githubusercontent.com/8658164/122102379-4310c000-ce0d-11eb-973f-96005d3e547c.png" width="300px">

pmusolino committed:
fix: discount rate

pmusolino committed:
fix: shipping label order summary with RTL languages

pmusolino commented:
Thank you @rachelmcr for your review.

> There's a small issue with the WooCommerce discount showing as a positive value when a signature option isn't selected for the shipping rate (see suggested code change below).

Fixed 👍 

> I also noticed a couple issues with accessibility and localization, but we could probably address those separately. (We should probably add a task to the Shipping Labels M3 project to test things like accessibility and localization through the whole flow; what do you think?)

We already have an issue opened about the Shipping Label Form with large font sizes: https://github.com/woocommerce/woocommerce-ios/issues/4343 we will fix it all together when we will work on this issue. 👍 

About the RTL languages issue, good catch, I fixed it 🙌 

<img src="https://user-images.githubusercontent.com/495617/122192741-3aaf9800-ce94-11eb-9745-edd6c1a4e310.png" width=300 />

rachelmcr reviewed (approved):
Looks good! :shipit:
